packages/rpc-graphql/src/loaders/__tests__/account-loader-test.ts

```diff
@@ -66,4 +66,1085 @@ describe('account loader', () => {
+                });
+                const getMultipleAccounts = jest
+                    .fn()
+                    .mockImplementationOnce(() => ({
+                        send: async () =>
+                            getMultipleAccountsMockResponse([
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: '11111111111111111111111111111111',
+                                },
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
+                                },
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'Stake11111111111111111111111111111111111111',
+                                },
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'Vote111111111111111111111111111111111111111',
+                                },
+                            ]),
+                    }))
+                    .mockImplementationOnce(() => ({
+                        send: async () =>
+                            getMultipleAccountsMockResponse([
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'NativeLoader1111111111111111111111111111111',
+                                },
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'BPFLoader2111111111111111111111111111111111',
+                                },
+                                {
+                                    data: ['AA', 'base64'],
+                                    owner: 'NativeLoader1111111111111111111111111111111',
+                                },
+                            ]),
+                    }));
+
+                // Then we should see `getAccountInfo` used for the single
+                // account in the last layer
+                const getAccountInfo = jest.fn().mockImplementationOnce(() => ({
+                    send: async () => ({
+                        context: {
+                            slot: 0,
+                        },
+                        value: defaultSpaceLamports({
+                            data: ['AA', 'base64'],
+                            owner: 'NativeLoader1111111111111111111111111111111',
+                        }),
+                    }),
+                }));
+
+                rpc = {
+                    getAccountInfo,
+                    getBlock: jest.fn(),
+                    getMultipleAccounts,
+                    getProgramAccounts: jest.fn(),
+                    getTransaction: jest.fn(),
+                };
```

Comment:
You don't have to recreate and regenerate the `rpcGraphQL` inside the test. You can ‘reach back’ into the existing resolvers and make them do new stuff.

```ts
it('coalesces multi-layered multiple requests into `getMultipleAccounts` requests', () => {
  rpc.getMultipleAccounts // <- already a mock function
    .mockReturnValue({
      send: jest.fn()
        .mockReturnValueOnce(/* first return value */)
        .mockReturnValueOnce(/* second return value */),
    });
});
```

Or you could do this work ahead of time.

```ts
let getMultipleAccountsSender;
beforeEach(() => {
  getMultipleAccountsSender = jest.fn();
  rpc = {
    getMultipleAccounts: jest.fn()
      .mockReturnValue({send: getMultipleAccountsSender}),
  };
});
it('coalesces multi-layered multiple requests into `getMultipleAccounts` requests', () => {
  rpc.getMultipleAccountsSender
    .mockReturnValueOnce(/* first return value */)
    .mockReturnValueOnce(/* second return value */);
});
```

Reply:
Done!